Source/WebKit/NetworkProcess/soup/NetworkDataTaskSoup.cpp

```diff
@@ -156,14 +161,59 @@ std::vector<std::string> tokenize(const std::string& s, const char* delimiters)
     return result;
 }
 
-void maybeSetUpClientAuthCertificates() {
-    if (!clientAuthCertificates().empty())
-        return;
+void maybeUpdateClientAuthCertificates(String& listURLs)
+{
+    if (!listURLs.isEmpty()) {
+        std::string urlsList(listURLs.utf8().data());
+        auto urls = tokenize(urlsList, ENV_URL_LIST_DELIMITER);
+        for (const auto& url : urls) {
+            String checkUrl = String::fromUTF8(url.c_str());
+            URL connectionURL(checkUrl);
+            auto foundCert = std::find_if(
+                clientAuthCertificates().begin(), clientAuthCertificates().end(),
+                [&connectionURL](const std::pair<std::string, GRefPtr<GTlsCertificate>>& p) {
+                    URL certURL({}, makeString(p.first.c_str()));
+                    return protocolHostAndPortAreEqual(certURL, connectionURL);
+                });
+
+            if (foundCert == clientAuthCertificates().end()) {
+                LOG(Network, "%s:%d Append new certificate for host %s\n", __func__, __LINE__, url.c_str());
+
+                auto* dataForUrl = getenv(url.c_str());
+                if (dataForUrl && *dataForUrl) {
+                    GUniqueOutPtr<GError> error;
+                    auto certificate = adoptGRef(
+                        g_tls_certificate_new_from_pem(dataForUrl,
+                            strlen(dataForUrl),
+                            &error.outPtr()));
+                    if (certificate) {
+                        clientAuthCertificates().insert(
+                            std::make_pair(url.c_str(), certificate));
+                    } else if (error) {
+                        WTFLogAlways("Error '%s' when trying to add client "
+                                     "certificate for %s\n",
+                            error->message, url.c_str());
+                    }
+                }
+            }
+        }
+    }
+}
 
-    auto* certsUrls = getenv("WPE_CLIENT_CERTIFICATES_URLS");
+void maybeSetUpClientAuthCertificates() {
+    auto* certsUrls = getenv(ENV_WPE_CLIENT_CERTIFICATES_URLS);
     if (certsUrls && *certsUrls) {
+        if (!clientAuthCertificates().empty()) {
+            String checkUrlsList = String::fromUTF8(certsUrls);
+            if(clientAuthCertificatesHash != checkUrlsList.hash()) {
+                maybeUpdateClientAuthCertificates(checkUrlsList);
+            } else {
+                return;
+            }
+        }
+
         std::string urlsList(certsUrls);
-        auto urls = tokenize(urlsList, "|");
+        auto urls = tokenize(urlsList, ENV_URL_LIST_DELIMITER);
```

Comment:
would be less confusing with the std::string named urls and std::vector\<std::string\> named urlsList

Reply:
This commit does not introduce the type _auto_, so I would prefer to leave it as is

Reply:
This is not what I meant. I mean rename the string from urlsList to urls, and the vector from urls to urlsList.
But either way, I see the names were like that before. You can leave it as is.

Reply:
In this line, I just updated the delimiter to use a variable ENV_URL_LIST_DELIMITER, and since this function is not updated I would prefer not to introduce more changes.